pyglider/utils.py

```diff
@@ -360,6 +360,25 @@ def nmea2deg(nmea):
            np.sign(nmea) * np.remainder(np.abs(nmea), 100) / 60)
     return deg
 
+
+def oxygen_concentration_correction(data, ncvar):
+    oxy_yaml = ncvar['oxygen_concentration']
+    if 'reference_salinity' not in oxy_yaml.keys():
+        _log.warning('No reference_salinity found in oxygen deployment yaml. Assuming reference salinity of 0 psu')
+        ref_sal = 0
+    else:
+        ref_sal = float(oxy_yaml['reference_salinity'])
+    sa = gsw.SA_from_SP(data['salinity'], data['pressure'], data['longitude'], data['latitude'])
+    ct = gsw.CT_from_t(sa, data['temperature'], data['pressure'])
+    o2_sol = gsw.O2sol(sa, ct, data['pressure'], data['longitude'], data['latitude'])
+    o2_sat = data['oxygen_concentration'] / gsw.O2sol(sa*0 + ref_sal, data['temperature'], data['pressure']*0, data['longitude'], data['latitude'])
+    data['oxygen_concentration'].values = o2_sat * o2_sol
+    data['oxygen_concentration'].attrs['comment'] = f'oxygen concentration corrected for salinity using gsw.02sol and ' \
```

Comment:
Is "comment" the right attribute here?  Do we have a "processing steps" attribute?  What does OG 1.0 say about detailing processing?  If we do use "comment" do we need to be careful to preserve any previous content of "comment" if it exists?

Reply:
Good point. If we use comment, we should append rather than replacing.

As for OG 1.0, it looks like we should use ocygen_concentration_QC:RTQC_methodology 

https://github.com/OceanGlidersCommunity/OG-format-user-manual/blob/main/OG_Format.adoc#geophysical-variables

One should also link to the QC methodology. Guess I'd better read that oxygen SOP

Reply:
Reference for salinity correction is to Gacia & Gordon 1992 https://aslopubs.onlinelibrary.wiley.com/doi/abs/10.4319/lo.1992.37.6.1307

And the argo oxygen processing manual

https://archimer.ifremer.fr/doc/00287/39795/

Reply:
That seems better than "comment'.  I'm not so concerned that the contents of this are correct, or even that the attribute name is the final one.  Just if we use "comment" we'd not want to lose previous/other comments, and that perhaps that is too general.  So I'm fine with whatever we use in here and we can always refine later.